openedx/core/djangoapps/credit/api.py

```diff
@@ -0,0 +1,156 @@
+                        "name": "midterm",
+                        "criteria": {},
+                    },
+                    {
+                        "namespace": "proctored_exam",
+                        "name": "final",
+                        "criteria": {},
+                    },
+                    {
+                        "namespace": "grade",
+                        "name": "grade",
+                        "criteria": {"min_grade": 0.8},
+                    },
+                ])
+
+    Raises:
+        InvalidCreditRequirements
+
+    Returns:
+        None
+    """
+
+    invalid_requirements = _validate_requirements(requirements)
+    if invalid_requirements:
+        invalid_requirements = ", ".join(invalid_requirements)
+        raise InvalidCreditRequirements(invalid_requirements)
+
+    credit_course, created = CreditCourse.get_or_create_credit_course(course_key=course_key)
+    if not created:
+        old_requirements = CreditRequirement.get_course_requirements(course_key=course_key)
+        requirements_to_disable = _get_requirements_to_disabled(old_requirements, requirements)
+        if requirements_to_disable:
+            CreditRequirement.disable_credit_requirements(requirements_to_disable)
+
+    for requirement in requirements:
+        CreditRequirement.add_or_update_course_requirement(credit_course, requirement)
+
+
+def get_credit_requirements(course_key, namespace=None):
+    """ Returns the requirements of a given course and namespace
+
+    Args:
+        course_key(CourseKey): The identifier for course
+        namespace(str): Namespace of requirements
+
+    Example:
+        >>> get_credit_requirements("course-v1-edX-DemoX-1T2015")
+                {
+                    requirements =
+                    [
+                        {
+                            "namespace": "verification",
+                            "name": "verification",
+                            "criteria": {},
+                        },
+                        {
+                            "namespace": "reverification",
+                            "name": "midterm",
+                            "criteria": {},
+                        },
+                        {
+                            "namespace": "proctored_exam",
+                            "name": "final",
+                            "criteria": {},
+                        },
+                        {
+                            "namespace": "grade",
+                            "name": "grade",
+                            "criteria": {"min_grade": 0.8},
+                        },
+                    ]
+                }
+
+    Returns:
+        Dict of requirements in the given namespace
+    """
+
+    requirements = CreditRequirement.get_course_requirements(course_key, namespace)
+    return [
+        {
+            "namespace": requirement.namespace,
+            "name": requirement.name,
+            "criteria": requirement.configuration
```

Comment:
I made a typo in the design document: the model field should be called `criteria` so it's consistent with the API.